iam_ltk.go

```diff
@@ -111,7 +112,25 @@ func (c *Client) GetLongTermKeys() (*GetLongTermKeysResponse, error) {
 func (c *Client) GetLongTermKey(iamUsername string) (*GetLongTermKeyResponse, error) {
 	log.Printf("[INFO] Getting long term key")
 
-	req, err := c.NewRequest(nil, "GET", "/ltk/search/"+iamUsername)
+	var req *http.Request
+	var err error
+
+	accountID, err := c.AccountDetails.GetAccountNumber()
+	if err != nil {
+		return nil, fmt.Errorf("error reading Account value: %s", err)
+	}
+
+	roleName, err := c.AccountDetails.GetRoleName(false)
+	if err != nil {
+		return nil, fmt.Errorf("error reading Role value: %s", err)
+	}
```

Comment:
You should only call these two methods if we aren't using STS credentials.  _Technically_ they might not be populated.